core/src/main/java/com/statful/client/core/sender/BufferedMetricsSender.java

```diff
@@ -316,4 +323,38 @@ private String buildAggregatedUri(final Aggregation aggregation, final Aggregati
 
         return buffersAsList;
     }
+
+    private void scheduleTask(final Runnable runnable) {
+        boolean wasScheduled = false;
+        if (executorService.getQueue().size() < configuration.getMaxWorkerTasksQueueSize()) {
```

Comment:
Since `futuresQueue` is an `ArrayBlockingQueue` we could just offer the future since according to the signature it will return `false` if the queue is full.